packages/bff/src/routers/catalogRouter.ts

```diff
@@ -78,19 +78,72 @@ const catalogRouter = (
     )
     .post(
       "/eservices/:eServiceId/descriptors/:descriptorId/activate",
-      async (_req, res) => res.status(501).send()
+      async (req, res) => {
+        const ctx = fromBffAppContext(req.ctx, req.headers);
+        try {
+          await catalogService.activateDescriptor(
+            req.params.eServiceId,
+            req.params.descriptorId,
+            ctx
+          );
+          return res.status(204).json().send();
+        } catch (error) {
+          const errorRes = makeApiProblem(error, emptyErrorMapper, ctx.logger);
+          return res.status(errorRes.status).json(errorRes).end();
+        }
+      }
     )
     .post(
       "/eservices/:eServiceId/descriptors/:descriptorId/update",
-      async (_req, res) => res.status(501).send()
+      async (req, res) => {
+        const ctx = fromBffAppContext(req.ctx, req.headers);
+        try {
+          const { id } = await catalogService.updateDescriptor(
```

Comment:
```suggestion
          const response = await catalogService.updateDescriptor(
```

Then you can directly `.send(response)` below instead of destructuring and rebuilding the object.